internal/agent/app/app.go

```diff
@@ -111,15 +115,58 @@ func (a *AgentApp) Run(ctx context.Context) {
 		case <-pollTicker.C:
 			a.poller.Update()
 		case <-reportTicker.C:
-			func() {
-				for metricName, metricValue := range a.poller.Metrics.Gauge {
-					a.client.UpdateJSONGauge(metricName, metricValue)
-				}
-				for metricName, metricValue := range a.poller.Metrics.Counter {
-					a.client.UpdateJSONCounter(metricName, metricValue)
-				}
-				a.poller.FlushPollCount()
-			}()
+			a.batchReport()
+			// a.simpleReport()
+		}
+	}
+}
+
+func (a *AgentApp) simpleReport() {
+	defer a.poller.FlushPollCount()
+	for metricName, metricValue := range a.poller.Metrics.Gauge {
+		if err := a.client.UpdateJSONGauge(metricName, metricValue); err != nil {
+			a.logger.Sugar().Errorf("failed update gauge metric %s: %v", metricName, err)
+		}
+	}
+	for metricName, metricValue := range a.poller.Metrics.Counter {
+		if err := a.client.UpdateJSONCounter(metricName, metricValue); err != nil {
+			a.logger.Sugar().Errorf("failed update gauge metric %s: %v", metricName, err)
+		}
+	}
+}
+
+func (a *AgentApp) batchReport() {
+	defer a.poller.FlushPollCount()
+	var metrics []*httpModels.Metric
```

Comment:
```suggestion
	metrics := make([]*httpModels.Metric, 0, len(a.poller.Metrics.Gauge) + len(a.poller.Metrics.Counter))
```
https://t.me/bushigo/11
https://t.me/bushigo/12